v2/pkg/variablesvalidation/variablesvalidation.go

```diff
@@ -87,6 +89,11 @@ const (
 	pathItemKindArray
 )
 
+func devModeCheck() bool {
+	d := strings.ToLower(os.Getenv("DEV_MODE"))
```

Comment:
Can we turn this into a property on the variablesVisitor that we can pass on creation?
This way, it won't depend on env vars, etc. which makes it more universally usable.

Reply:
Hey @jensneuse , thanks for the suggested change. @Sam-tesouro and I were just chatting, and I'm afraid my go foo is a bit rusty so wondering how we'd pass that into the variables visitor. If you have a quick example, suggestion, or link, that'd help. We're working on it either way though. Thanks for all you do!

Reply:
Implemented a thing that might be what you want, possibly.

Reply:
I see the issue with being so coupled to the DEV_MODE env var, working on decoupling still. Biggest hangup currently is just figuring out how I am going to achieve test coverage.

Reply:
Thank you! I believe I achieved decoupling. Glad to make further modifications if you have preferences!